packages/react/src/components/AccountSettings/ConfigureTOTP/ConfigureTOTP.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import QRCode from 'qrcode';
+
+import {
+  setupTOTP,
+  verifyTOTPToken,
+  getLogger,
+  translate,
+  getTotpCodeURL,
+  AmplifyUser,
+} from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { View, Flex } from '../../../primitives';
+import { FormValues } from '../types';
+import {
+  ConfirmationCode,
+  CopySecretKey,
+  Error,
+  SecretKeyQRCode,
+  SubmitButton,
+} from './defaults';
+import { ConfigureTOTPProps } from './types';
+
+const logger = getLogger('Auth');
+
+function ConfigureTOTP({
+  totpIssuer = 'AWSCognito',
+  totpUsername,
+  onSuccess,
+  onError,
+}: ConfigureTOTPProps): JSX.Element | null {
+  const [secretKey, setSecretKey] = React.useState<string>(null);
+  const [qrCode, setQrCode] = React.useState<string>(null);
+  const [formValues, setFormValues] = React.useState<FormValues>({ code: '' });
+  const [errorMessage, setErrorMessage] = React.useState<string>(null);
+
+  const { user, isLoading } = useAuth();
+
+  const hasInit = React.useRef(false);
+
+  const generateQRCode = React.useCallback(
+    async (user: AmplifyUser): Promise<void> => {
+      try {
+        const newSecretKey = await setupTOTP(user);
+        const username = totpUsername || user?.username;
+        const totpCode = getTotpCodeURL(totpIssuer, username, newSecretKey);
+        const qrCodeImageSource = await QRCode.toDataURL(totpCode);
+
+        if (!hasInit.current) {
+          setSecretKey(newSecretKey);
+          setQrCode(qrCodeImageSource);
+          hasInit.current = false;
```

Comment:
Should this be setting `hasInit.current` to `true`?

Reply:
100%, this is a typo from my end. [b48ab69](https://github.com/aws-amplify/amplify-ui/pull/2969/commits/b48ab6988dbae259e3ed25e3a0dc3bc4d04bbec9)